jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -1865,47 +2136,109 @@ public VarHandle unreflectVarHandle(Field field) throws IllegalAccessException {
 			 * as it is not set up for lookup objects by default.
 			 */
 			int fullAccessMode = FULL_ACCESS_MASK | MODULE | UNCONDITIONAL;
+			
+			switch(dropMode) {
+			case PUBLIC:
+			case MODULE:
+			case PACKAGE:
+			case PRIVATE:
+			case PROTECTED:
+			case UNCONDITIONAL:
+				/* dropMode is OK */
+				break;
+			default:
+				/*[MSG "K065R", "The requested lookup mode: 0x{0} is not one of the existing access modes: 0x{1}"]*/
+				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K065R", Integer.toHexString(dropMode), Integer.toHexString(fullAccessMode))); //$NON-NLS-1$
+			}
 
+			/*[IF Java14]*/
+			/* The lookup object has to discard the protected access by default */
+			int newAccessMode = accessMode & ~PROTECTED;
+			/*[ELSE]*/
 			/* The lookup object has to discard the protected and unconditional access by default */
 			int newAccessMode = accessMode & ~(PROTECTED | UNCONDITIONAL);
+			/*[ENDIF] Java14*/
 			
-			switch (dropMode) {
+			/* The access mode to be dropped must exist in the current access mode;
+			 * otherwise, the new access mode remains unchanged.
+			 */
+			switch (dropMode & newAccessMode) {
 			case PUBLIC:
 				newAccessMode = NO_ACCESS;
 				break;
-			case MODULE:
-				newAccessMode &= ~(MODULE | PACKAGE | PRIVATE);
-				break;
 			case PACKAGE:
 				newAccessMode &= ~(PACKAGE | PRIVATE);
 				break;
 			case PRIVATE:
 				newAccessMode &= ~PRIVATE;
 				break;
-			case PROTECTED:
 			case UNCONDITIONAL:
+				/*[IF Java14]*/
+				newAccessMode = NO_ACCESS;
+				/*[ENDIF] Java14*/
 				break;
 			default:
-				/*[MSG "K065R", "The requested lookup mode: 0x{0} is not one of the existing access modes: 0x{1}"]*/
-				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K065R", Integer.toHexString(dropMode), Integer.toHexString(fullAccessMode))); //$NON-NLS-1$
+				/* no change in the access mode */
 			}
 			
-			return new Lookup(accessClass, newAccessMode); 
+			/* The exception is MODULE in which case all access bits involved must be dropped
+			 * whether or not the MODULE bit exists in the access mode.
+			 */
+			if ((dropMode == MODULE) || ((dropMode & newAccessMode) == MODULE)) {
+				newAccessMode &= ~(MODULE | PACKAGE | PRIVATE);
+			}
+			
+			/*[IF Java14]*/
+			/* There is no previous lookup class for the requested lookup class
+			 * if the MODULE or UNCONDITIONAL bit is set in the new access mode.
+			 */
+			Class<?> newPrevAccessClass = prevAccessClass;
+			if (((newAccessMode & MODULE) == MODULE)
+				|| ((newAccessMode & UNCONDITIONAL) == UNCONDITIONAL)
+			) {
+				newPrevAccessClass = null;
+			}
```

Comment:
at http://cr.openjdk.java.net/~iris/se/14/build/latest/api/java.base/java/lang/invoke/MethodHandles.Lookup.html#UNCONDITIONAL:
```
public static final int UNCONDITIONAL
...
If this lookup mode is set, the previous lookup class is always null. 
<----- this is the new change when setting up UNCONDITIONAL since Java 14
```
The newly added description requires to handle `previousAccessClass` which is not specific to `dropLookupMode()` but everywhere when the UNCONDITIONAL bit is set initially.